### ISSUE TITLE
Slow page load due to sequential Infura requests

### ISSUE THREAD

**Issue by jakubgs** [dap-ps/discover#54]
Based on page loading analysis from https://www.webpagetest.org/ we can see that the sequentially made Infura requests made after the `/metadata/all` call are the main cause of slow page loads:
![dap_ps_infura_slow](https://user-images.githubusercontent.com/2212681/68328095-ed235480-00ce-11ea-8bfb-4eca3f123d8b.png)
https://www.webpagetest.org/result/191106_VA_1baab60b0d5daa3a4ee44b6e8ca482fb/1/details/#step1_request29
This delays the first load of page contents for at least 4 seconds on the first page load.

**Comment by jakubgs** [comment]
Pretty sure this is the part that causes that:
https://github.com/dap-ps/discover/blob/bf8b9bcbad0a4dec550a9d98a27fb62f7b8a38ce/src/common/blockchain/services/discover-service/discover-service.js#L59-L67

**Comment by jakubgs** [comment]
Here's an example solution:
```javascript
let asyncCalls = [...Array(contractDappsCount).keys()].map(
  i => DiscoverContract.methods.dapps(i).call({ from: this.sharedContext.account })
)
/* using Promise.all() to run calls in parallel */
let dapps = await Promise.all(asyncCalls)

for (dapp of dapps) {
  /* handle the returned dapps */
}
```
But not sure if it would work. I wanted to test it on https://dev.dap.ps/ but [it seems broken](https://github.com/dap-ps/infra-dapps/issues/16).

**Comment by jakubgs** [comment]
Here's the same part as in issue description but after deploying the fix:
![dapps_new_parallel_load](https://user-images.githubusercontent.com/2212681/70747943-4d8a5f00-1d29-11ea-9a38-aef2712fc912.png)

**Comment by jakubgs** [comment]
I had to add a small fix after that: ccae507c00f2292a93a8819ce60618d9cfd12dab
Result:
![dapps_new_parallel_load_2](https://user-images.githubusercontent.com/2212681/70756879-0eb2d400-1d3e-11ea-8c00-f752f4fcd032.png)
